### PR TITLE
ARROW-7375: [Python] Expose C++ MakeArrayOfNull

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by alippai** [comment]
Can this be extended to support any scalar value? Creating a column with single value is a common step for us (before concatenating tables, so the fragments a are differentiated in the result)

**Comment by wesm** [comment]
@alippai that is doable but would need to get done in a separate PR

File: python/pyarrow/__init__.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def parse_git(root, **kwargs):
                          schema,
                          unify_schemas,
                          Array, Tensor,
-                         array, chunked_array, record_batch, table,
+                         array, chunked_array, record_batch, table, nulls,
```
Comment (wesm) [inline python/pyarrow/__init__.py:93]: Can you add it to the API listing .rst page?

**Review by wesm — APPROVED** [review]
+1 modulo adding `pyarrow.nulls` to the API listing

File: python/pyarrow/__init__.py (hunk 1 above)
Comment (kszucs) [reply python/pyarrow/__init__.py:93]: Sure.

**Pull request closed without merging**